helpers/stable_swap_math/mod.rs

```diff
@@ -66,19 +69,21 @@ fn compute_d(amounts: &Vec<u128>, amp_coef: u128) -> Result<U256, MathError> {
         // Computes next D unitl satisfying precision is reached
         for _ in 0..MAX_ITERATIONS {
             let d_next = compute_d_next(d, n, amounts, ann_sum, ann_sub_one, n_add_one)?;
-            if d_next > d {
-                if d_next.checked_sub(d).ok_or(MathError::SubUnderflow(2))? <= 1.into() {
-                    return Ok(d);
-                }
-            } else if d.checked_sub(d_next).ok_or(MathError::SubUnderflow(3))? <= 1.into() {
+            // [nit] Maybe like this?
+            if d_next.abs_diff(d) <= 1.into() {
                 return Ok(d);
             }
             d = d_next;
         }
+        // [audit] Shouldn't we return precision error here?
```

Comment:
That's a good point. It was implemented this way so it would always be possible for LPs to withdraw, like in [Ref](https://github.com/ref-finance/ref-contracts/blob/d241d7aeaa6250937b160d56e5c4b5b48d9d97f7/ref-exchange/src/stable_swap/math.rs#L179) implementation. Later, `withdraw_liquidity_by_share(...)` was added which allows withdrawing without calculating invariant so it makes sense to return an error here.